techniques/systemSettings/misc/clockConfiguration/4.0/clockConfiguration.st

```diff
@@ -0,0 +1,462 @@
+
+      &if(CLOCK_FQDNNTP)&
+      "keytemp" string => join("\,0x1 ", ntpServers);
+      "serverskey" string => "${keytemp}\,0x1";
+      &else&
+      "serverskey" string => join(" ", ntpServers);
+      &endif&
+
+      "target_windows_timezone" string => "${tz_variables.tz_win[${target_timezone}]}";
+      "current_windows_timezone" string => execresult("\"${sys.winsysdir}\tzutil.exe\" /g", "useshell");
+
+    linux::
+      "linux_timezone"   string => "${tz_variables.tz_linux[${target_timezone}]}";
+
+  classes:
+
+      "clock_timezone_nochange"  expression => strcmp("&CLOCK_TIMEZONE&","dontchange");
+      "clock_timezone_valid" expression => isvariable("tz_variables.tz_linux[${target_timezone}]");
+      "clock_timezone_noedit" expression => "clock_timezone_nochange|!clock_timezone_valid";
+
+      "clock_hwclock_sync" expression => strcmp("&CLOCK_HWSYNC_ENABLE&","true");
+      "clock_date_s_now" expression => strcmp("&CLOCK_DATE_S_NOW_ENABLE&","true");
+
+      "clock_hwclock_has_rtc" expression => fileexists("/dev/rtc");
+
+      "ntp_config_file_exists" expression => fileexists("/etc/ntp.conf");
+
+    clock_timezone_valid.windows::
+      # check if we need to change the windows timezone
+      "need_to_change_timezone" not => strcmp("${target_windows_timezone}", "${current_windows_timezone}");
+      "clock_hwclock_rtc2utc" expression => strcmp("&CLOCK_HWSYNC_RTC&", "UTC");
+
+    any::
+      "pass3" expression => "pass2";
+      "pass2" expression => "pass1";
+      "pass1" expression => "any";
+
+  files:
+
+    # Adjust ntp.conf (Add the servers)
+    !windows.(ntp_config_file_exists|ntp_installed)::
+      "/etc/ntp.conf"
+        edit_line => setNtpServer("@{this.ntpServers}"),
+        edit_defaults => noempty_backup,
+        classes => kept_if_else("ntpconf_kept", "repaired_ntpconf", "not_repaired_ntpconf");
+
+    # Copy the correct localtime file (distro independant)
+    !clock_timezone_noedit.!windows.clock_timezone_valid::
+      "/etc/localtime"
+        copy_from => digest_cp("/usr/share/zoneinfo/${linux_timezone}"),
+        comment => "Updating the /etc/localtime file",
+        classes => kept_if_else("localtime_kept", "repaired_localtime", "not_repaired_localtime");
+
+    # Edit the distro dependant files to set the TZ on boot
+    (redhat.!clock_timezone_noedit).clock_timezone_valid::
+      "/etc/sysconfig/clock"
+        edit_line => EditCentOSTimezone("${linux_timezone}", "${clock_hwclock_rtc}"),
+        edit_defaults => noempty_backup,
+        classes => kept_if_else("redhattz_kept", "repaired_redhattz", "not_repaired_redhattz");
+
+    (SuSE.!clock_timezone_noedit).clock_timezone_valid::
+      "/etc/sysconfig/clock"
+        edit_line => EditSuSETimezone("${linux_timezone}", "${clock_hwclock_rtc}"),
+        edit_defaults => noempty_backup,
+        classes => kept_if_else("susetz_kept", "repaired_susetz", "not_repaired_susetz");
+
+    ## TODO: Handle UTC/Localtime on Debian
```

Comment:
Depends on the version. I did some googling:
https://www.kariliq.nl/misc/debian-clock.html
But for my approach missing this is not worse then the previous version of the technique, where it was also not handled... This would require some research and testing for debian/Ubuntu which I had no time for...